File: HTTPShortcuts/app/src/main/java/ch/rmy/android/http_shortcuts/icons/IconAdapter.java
package ch.rmy.android.http_shortcuts.icons;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.ViewGroup;

import ch.rmy.android.http_shortcuts.listeners.OnIconSelectedListener;

class IconAdapter extends RecyclerView.Adapter<IconViewHolder> {

    private final Context context;
    private final OnIconSelectedListener listener;

    protected IconAdapter(Context context, OnIconSelectedListener listener) {
        this.context = context;
        this.listener = listener;
    }

    @Override
    public IconViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new IconViewHolder(context, parent, listener);
    }

    @Override
    public void onBindViewHolder(IconViewHolder holder, int position) {
        holder.setIcon(Icons.ICONS[position]);
    }

    @Override
    public int getItemCount() {
        return Icons.ICONS.length;
    }
}

File: HTTPShortcuts/app/src/main/kotlin/ch/rmy/android/http_shortcuts/icons/IconAdapter.kt
package ch.rmy.android.http_shortcuts.icons

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import ch.rmy.android.http_shortcuts.databinding.IconListItemBinding

class IconAdapter(
    private val icons: List<ShortcutIcon>,
    private val listener: (ShortcutIcon) -> Unit,
) : RecyclerView.Adapter<IconViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): IconViewHolder =
        IconViewHolder(IconListItemBinding.inflate(LayoutInflater.from(parent.context), parent, false), listener)

    override fun onBindViewHolder(holder: IconViewHolder, position: Int) {
        holder.setIcon(icons[position])
    }

    override fun getItemCount() = icons.size

}